lib/models/ui.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const UI = require('console-ui');
+
+let singletonUI;
+
+/**
+ * The class that serves as a singleton wrapper for `console-ui`.
+ *
+ * @private
+ * @module ember-cli
+ * @class CommandLineUI
+ * @constructor
+ */
+class CommandLineUI {
+  constructor(args) {
+    let options = args || {};
+
+    if (!singletonUI) {
+      singletonUI = new UI({
+        inputStream: options.inputStream,
+        outputStream: options.outputStream,
+        errorStream: options.errorStream || process.stderr,
+        errorLog: options.errorLog || [],
+        ci: process.env.CI || (/^(dumb|emacs)$/).test(process.env.TERM),
```

Comment:
I believe if terminal has no ANSI support then `TERM` will be set to `dumb` (meaning there will be no colours); it is sort of a "last resort, when everything is broken" kind of thing.